frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -182,18 +198,42 @@ impl CallDef {
 					return Err(syn::Error::new(method.sig.span(), msg))
 				}
 
-				let mut call_var_attrs: Vec<FunctionAttr> =
-					helper::take_item_pallet_attrs(&mut method.attrs)?;
-
-				if call_var_attrs.len() != 1 {
-					let msg = if call_var_attrs.is_empty() {
+				let (mut weight_attrs, mut call_idx_attrs): (Vec<FunctionAttr>, Vec<FunctionAttr>) =
+					helper::take_item_pallet_attrs(&mut method.attrs)?.into_iter().partition(
+						|attr| {
+							if let FunctionAttr::Weight(_) = attr {
+								true
+							} else {
+								false
+							}
+						},
+					);
+
+				if weight_attrs.len() != 1 {
+					let msg = if weight_attrs.is_empty() {
 						"Invalid pallet::call, requires weight attribute i.e. `#[pallet::weight($expr)]`"
 					} else {
 						"Invalid pallet::call, too many weight attributes given"
 					};
 					return Err(syn::Error::new(method.sig.span(), msg))
 				}
-				let weight = call_var_attrs.pop().unwrap().weight;
+				let weight = match weight_attrs.pop().unwrap() {
+					FunctionAttr::Weight(w) => w,
+					_ => unreachable!("checked during creation of the let binding"),
+				};
+
+				if call_idx_attrs.len() > 1 {
+					let msg = "Invalid pallet::call, too many call_index attributes given";
+					return Err(syn::Error::new(method.sig.span(), msg))
+				}
+				let call_index = call_idx_attrs.pop().map(|attr| match attr {
+					FunctionAttr::CallIndex(idx) => idx,
+					_ => unreachable!("checked during creation of the let binding"),
+				});
+
+				let final_index =
+					call_index.unwrap_or_else(|| last_index.map(|idx| idx + 1).unwrap_or(0));
+				last_index = Some(final_index);
```

Comment:
I don't see this being used henceforth, why do we assign to it?

Reply:
It's being used during expansion to fill up the value in the `#[codec(index = $idx)]` attribute.